Cython/Compiler/ExprNodes.py

```diff
@@ -8179,10 +8182,21 @@ def generate_sequence_packing_code(self, code, target=None, plain=False):
                 else:
                     size_factor = ' * (%s)' % (c_mult,)
 
-        if self.type is tuple_type and (self.is_literal or self.slow) and not c_mult:
+        if ((self.type is tuple_type or self.type is list_type) and
+                (self.is_literal or self.slow) and
+                not c_mult and
+                len(self.args) > 0):
```

Comment:
There seems to be a related special case in the `generate_subexpr_disposal_code()` method further down.
Since we're implementing the list helper function ourselves, we could also consider letting it steal the item references. But that's an optimisation, not required here.